frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -811,6 +811,22 @@ function ReaderDictionary:startSdcv(word, dict_names, fuzzy_search)
         end
     end
 
+    -- If every word contains a CJK character, every word candidate is
+    -- (probably) a CJK word. We don't want fuzzy searching in this case
+    -- because sdcv cannot handle CJK text properly when fuzzy searching (with
+    -- Japanese, it returns hundreds of useless results).
+    local shouldnt_fuzzy_search = true
+    for _, word in ipairs(words) do
+        if not util.hasCJKChar(word) then
+            shouldnt_fuzzy_search = false
```

Comment:
```lua
    local shouldnt_fuzzy_search = false
    for _, word in ipairs(words) do
        if util.hasCJKChar(word) then
            shouldnt_fuzzy_search = true
```
Would have been simpler to read

Reply:
That is not the same logic. This PR only disables fuzzy searching if *every* word in the array has CJK characters. Your suggestion would disable fuzzy searching if *any* word has CJK characters. In practice they might always give the same results, but it's not the same logic.